aptos-move/framework/aptos-framework/sources/voting.move

```diff
@@ -299,6 +309,9 @@ module aptos_framework::voting {
         let proposal = table::borrow_mut(&mut voting_forum.proposals, proposal_id);
         assert!(!proposal.is_resolved, error::invalid_state(EPROPOSAL_ALREADY_RESOLVED));
 
+        let multi_step_key = utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        assert!(!simple_map::contains_key(&proposal.metadata, &multi_step_key) || *simple_map::borrow(&proposal.metadata, &multi_step_key) == SINGLE_STEP_PROPOSAL, error::unauthenticated(EPROPOSAL_IS_NOT_SINGLE_STEP));
```

Comment:
Any particular reason why we have one table but with two types of keys instead of two tables? Does that has anything to do with backwards compatibility?

Reply:
Yeah it is for backward compatibility; existing (single-step) proposals wouldn't have the `multi_step_key` in their `proposal.metadata`. So we would assume a proposal is single-step if their simple map does not have the `multi_step_key`, or if the value is `SINGLE_STEP_PROPOSAL` to their `multi_step_key`.

Reply:
Why not just use a boolean and interpret the lack of this flag as single-step? This should maintain backward compatibility

Reply:
ah so it's bc we  made the metadate simple map to be of type `SimpleMap<String, vector<u8>>` [here](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/aptos_governance.move#L414)

Reply:
btw we're interpreting the lack of this flag as single-step in the code above

Reply:
We can just convert back and forth between vector<u8> and bool right?

Reply:
oh yeah makes sense, I will use a boolean and convert it to vector<u8> then